src/main/java/org/opensearch/jobscheduler/transport/request/ExtensionJobActionRequest.java

```diff
@@ -27,7 +34,63 @@
      * @throws IOException if serialization fails
      */
     public ExtensionJobActionRequest(String extensionActionName, T actionParams) throws IOException {
-        super(extensionActionName, JobDetailsService.convertParamsToBytes(actionParams));
+        super(extensionActionName, convertParamsToBytes(actionParams));
+    }
+
+    /**
+     * Finds the index of the specified byte value within the given byte array
+     *
+     * @param bytes the byte array to process
+     * @param value the byte to identify index of
+     * @return the index of the byte value
+     */
+    private static int indexOf(byte[] bytes, byte value) {
+        for (int offset = 0; offset < bytes.length; ++offset) {
+            if (bytes[offset] == value) {
+                return offset;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Trims off the fully qualified request class name bytes and null byte from the ExtensionActionRequest requestBytes
+     *
+     * @param requestBytes the request bytes of an ExtensionActionRequest
+     * @return the trimmed array of bytes
+     */
+    public static byte[] trimRequestBytes(byte[] requestBytes) {
+        int nullPos = indexOf(requestBytes, ExtensionJobActionRequest.UNIT_SEPARATOR);
+        return Arrays.copyOfRange(requestBytes, nullPos + 1, requestBytes.length);
+    }
+
+    /**
+     * Converts an object of type T that extends {@link Writeable} into a byte array and prepends the fully qualified class name bytes
+     *
+     * @param <T> a class that extends writeable
+     * @param actionParams the action parameters to be serialized
+     * @throws IOException if serialization fails
+     * @return the byte array of the parameters
+     */
+    private static <T extends Writeable> byte[] convertParamsToBytes(T actionParams) throws IOException {
```

Comment:
I wonder if this should be a util method on ExtensionActionRequest or ExtensionTransportAction